samples/sid_end_device/src/sensor_monitoring/app.c

```diff
@@ -163,25 +164,58 @@ static void on_sidewalk_status_changed(const struct sid_status *status, void *co
 	}
 }
 
-static void sidewalk_btn_handler(uint32_t event)
+static void app_event_dfu_mode(sidewalk_ctx_t *sid, void *ctx)
 {
-	int err = sidewalk_event_send((sidewalk_event_t)event, NULL, NULL);
-	if (err) {
-		LOG_ERR("Send event err %d", err);
-		return;
-	};
-
-	if (SID_EVENT_NORDIC_DFU == event) {
-		static bool in_dfu;
-		if (in_dfu) {
-			in_dfu = false;
-			k_timer_start(&notify_timer, K_MSEC(NOTIFY_TIMER_DURATION_MS),
-				      K_MSEC(CONFIG_SID_END_DEVICE_NOTIFY_DATA_PERIOD_MS));
-		} else {
-			in_dfu = true;
-			k_timer_stop(&notify_timer);
+	bool *enter_dfu_state = (bool *)ctx;
+
+	int err = -ENOTSUP;
+	if (*enter_dfu_state) {
+		// go to DFU state
+		sidewalk_event_exit(sid, NULL);
+		k_timer_stop(&notify_timer);
+		LOG_INF("Entering into DFU mode");
+#if defined(CONFIG_SIDEWALK_DFU_SERVICE_BLE)
+		err = nordic_dfu_ble_start();
+#endif
+		if (err) {
+			LOG_ERR("dfu start err %d", err);
 		}
+	} else {
+		// Exit from DFU state
+#if defined(CONFIG_SIDEWALK_DFU_SERVICE_BLE)
+		err = nordic_dfu_ble_stop();
+#endif
+		if (err) {
+			LOG_ERR("dfu stop err %d", err);
+		}
+		sidewalk_event_autostart(sid, NULL);
+		k_timer_start(&notify_timer, K_MSEC(NOTIFY_TIMER_DURATION_MS),
+			      K_MSEC(CONFIG_SID_END_DEVICE_NOTIFY_DATA_PERIOD_MS));
+	}
+}
+
+static void app_btn_dfu_state(uint32_t unused)
+{
+	ARG_UNUSED(unused);
+	static bool go_to_dfu_state = true;
+	int e = sidewalk_event_send(app_event_dfu_mode, &go_to_dfu_state, NULL);
```

Comment:
consider split to dfu_mode_enter and _exit, insted of passing the state variable, this should simplify code